src/sisl/io/wannier90/seedname.py

```diff
@@ -297,69 +278,133 @@ def _r_wigner_seitz_weights(self):
+
+        if "geometry" not in kwargs:
+            # to ensure we get the correct orbital positions
+            kwargs["geometry"] = self.read_geometry()
+        
+        kwargs["cutoff"] = cutoff
+
+        for f in order:
+            H = getattr(self, f"_r_hamiltonian_{f.lower()}")(*args, **kwargs)
+            if H is not None:
+                return H
+
+        return None
+
+    def ArgumentParser(self, p=None, *args, **kwargs):
+        """Returns the arguments that is available for this Sile"""
+        newkw = Geometry._ArgumentParser_args_single()
+        newkw.update(kwargs)
+        return self.read_geometry().ArgumentParser(p, *args, **newkw)
+
+class xyzSileWannier90(SileWannier90):
+        
+    @sile_fh_open()
+    def read_geometry(self, lattice):
+        """Defered routine"""
 
-        # Rewind to ensure we can read the entire matrix structure
         self.fh.seek(0)
 
-        # Time of creation
-        self.readline()
+        nc = int(self.readline())
 
-        # Number of orbitals
-        no = int(self.readline())
-        if no != geometry.no:
-            raise ValueError(
-                f"{self.__class__.__name__}"
-                ".read_hamiltonian has found inconsistent number "
-                "of orbitals in _hr.dat vs the geometry. Remember to re-run Wannier90?"
-            )
+        # Comment
+        self.readline()
 
-        ws = self._r_wigner_seitz_weights()
+        na = 0
+        sp = [None] * nc
+        xyz = np.empty([nc, 3], np.float64)
+        for ia in range(nc):
+            l = self.readline().split()
+            sp[ia] = l.pop(0)
+            if sp[ia] == "X":
+                na = ia + 1
+            xyz[ia, :] = [float(k) for k in l[:3]]
 
-        # List for holding the Hamiltonian
-        Hsc = defaultdict(lambda: lil_matrix((geometry.no, geometry.no), dtype=dtype))
-        is_complex = np.iscomplexobj(dtype(1))
+        return Geometry(xyz[:na, :], atoms="H", lattice=lattice)
 
-        iws = -1
-        isc = [0, 0, 0]
-        while True:
-            l = self.readline()
-            if l == "":
-                break
+class hamSileWannier90(SileWannier90):
+    
+    def _r_wigner_seitz_weights(self):
+        # Number of Wigner-Seitz degeneracy points
+        npts = int(self.readline())
 
-            # Split here...
-            l = l.split()
+        ws = []
+        while len(ws) < npts:
+            # both formats uses 15 points per line,
+            # however, this should be usable if they decide to change
+            # the number of counts per line.
+            ws.extend(list(map(int, self.readline().split())))
 
-            # Get super-cell, row and column
-            isc[0], isc[1], isc[2], r, c = map(int, l[:5])
+        ws = 1.0 / _a.arrayd(ws)
+        return ws
+class tbSileWannier90(hamSileWannier90):
 
-            # Update index for degeneracy, if required
-            if r + c == 2:
-                iws += 1
+    @sile_fh_open()
```

Comment:
do `sile_fh_open(True)` and remove `self.fh.seek(0)`, this will force a complete re-read of the file.